api/src/util/utility/Connections.ts

```diff
@@ -0,0 +1,136 @@
+import { Config, DiscordApiErrors } from "@fosscord/util";
+import fetch from "node-fetch";
+import { URL } from "url";
+import crypto from "crypto";
+
+export interface ConnectionAuthCallbackSchema {
+	code: string;
+	friend_sync: boolean;
+	insecure: boolean;
+	state: string;
+}
+
+export interface GithubConnectionUserInfo {
+	login: string;
+	id: number;
+	node_id: string;
+	avatar_url: string;
+	gravatar_id: string;
+	url: string;
+	html_url: string;
+	followers_url: string;
+	following_url: string;
+	gists_url: string;
+	starred_url: string;
+	subscriptions_url: string;
+	organizations_url: string;
+	repos_url: string;
+	events_url: string;
+	received_events_url: string;
+	type: string;
+	site_admin: boolean;
+	name: string;
+	company: string;
+	blog: string;
+	location: string;
+	email: string;
+	hireable?: any;
+	bio: string;
+	twitter_username: string;
+	public_repos: number;
+	public_gists: number;
+	followers: number;
+	following: number;
+	created_at: Date;
+	updated_at: Date;
+	private_gists: number;
+	total_private_repos: number;
+	owned_private_repos: number;
+	disk_usage: number;
+	collaborators: number;
+	two_factor_authentication: boolean;
+	plan: {
+		name: string;
+		space: number;
+		collaborators: number;
+		private_repos: number;
+	};
+}
+
+// TODO: maybe make this more class-based?
+// TODO: should probably have some checks if client id and/or client secret are missing while connection is enabled.
```

Comment:
I agree with this, a more OOP approach would be more maintainable/easier to read/understand